appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -2063,7 +2073,12 @@ private void getGAVURLs() throws GlassFishException {
                     contextRoot = null; // use only once
                 }
 
-                deploymentURLsMap.put(defaultContext, artefactMapEntry.getValue());
+                URL artefactURL = artefactMapEntry.getValue();
+                String artefactName = artefactURL.getPath().substring(artefactURL.getPath().lastIndexOf('/') + 1);
+
```

Comment:
With [above comment](https://github.com/payara/Payara/pull/5071/files#r548858934) implemented - could new method be used here?

Reply:
Same code also in UberJarCreator class. Maybe make sense move this to utility class? Or such class already exists?